src/ls/Git_helper.ml

```diff
@@ -0,0 +1,81 @@
+(*diff unified format regex https://www.gnu.org/software/diffutils/manual/html_node/Detailed-Unified.html#Detailed-Unified
```

Comment:
This is mostly independent of the language server, so it should be moved out, e.g., in libs/
and actually there are a few places where we do git related stuff that we should merge together.
maybe we can start a libs/lib_git/ to gather such code.